src/utils/mod.rs

```diff
@@ -32,42 +36,43 @@ impl Multipolygon {
 // Users of this code must verify correctness for their application.
 // http://geomalgorithms.com/a03-_inclusion.html
 
-fn is_point_in_polygon(p: &Point, v: &[Point]) -> bool {
+fn is_point_in_polygon(p: Point, v: &[Point]) -> bool {
     let mut wn = 0;
     let mut i = v.len() - 1;
-    for j in 0 .. v.len() {
+    // FIXME: this is not the most efficient way to do this. Use this instead:
+    //        for (j, val) in v.iter().enumerate() { if val.y <= p.y { ...
```

Comment:
Why? because `v[j]` comes at a cost? Do you know a source where I can read up about that?
I don't actually need `j`, I just need the value of `v[j]`

Reply:
> iterators are more efficient - they don't require out-of-bounds checks on each access

Ah, I see